File: backend/app/schemas/demo_request.py
from datetime import date
from typing import Optional
from pydantic import EmailStr
from app.schemas.base import TimestampSchema, BaseSchema

class DemoRequestBase(BaseSchema):
    full_name: str
    work_email: EmailStr
    phone: Optional[str] = None
    company_name: str
    job_title: Optional[str] = None
    company_size: Optional[str] = None
    product_interest: Optional[str] = None
    preferred_date: Optional[date] = None
    message: Optional[str] = None

class DemoRequestCreate(DemoRequestBase):
    pass

class DemoRequest(DemoRequestBase, TimestampSchema):
    id: int
    is_processed: bool = False

File: frontend/src/models/demoRequest.ts
export interface DemoRequestFormValues {
	full_name: string;
	work_email: string;
	phone: string;
	company_name: string;
	job_title: string;
	company_size: string;
	product_interest: string;
	preferred_date: string;
	message: string;
}

export interface DemoRequestPayload {
	full_name: string;
	work_email: string;
	phone?: string;
	company_name: string;
	job_title?: string;
	company_size?: string;
	product_interest?: string;
	preferred_date?: string;
	message?: string;
}

export interface DemoRequestResponse extends DemoRequestPayload {
	id: number;
	is_processed: boolean;
	created_at: string;
	updated_at: string;
}
